src/app/clients/ExpandedClientDetails.tsx

```diff
@@ -17,20 +30,91 @@ const DeletedText = styled.div`
     flex-direction: row;
 `;
 
-const ExpandedClientDetails = async ({ clientId }: Props): Promise<React.ReactElement> => {
-    const expandedClientDetails = await getExpandedClientDetails(clientId);
+const ExpandedClientDetails: React.FC<Props> = ({ clientId, displayClientsParcels = false }) => {
+    const [clientDetails, setClientDetails] = useState<ExpandedClientData | null>(null);
+    const [clientParcelsDetails, setClientParcelsDetails] = useState<
+        ExpandedClientParcelDetails[] | null
+    >(null);
 
-    const expandedClientParcelsDetails = await getClientParcelsDetails(clientId);
+    const originalNotes = clientDetails?.notes ?? null;
 
-    return (
-        <>
-            {expandedClientDetails.isActive ? (
-                <DataViewer data={{ ...expandedClientDetails }} />
-            ) : (
-                <DeletedText>Client has been deleted.</DeletedText>
-            )}
-            <ClientParcelsTable parcelsData={expandedClientParcelsDetails} />
-        </>
+    const [notes, setNotes] = useState<string | null>(originalNotes);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+
+    const loadData = useCallback(() => {
+        (async () => {
+            setIsLoading(true);
+            setClientDetails(await getExpandedClientDetails(clientId));
+            setClientParcelsDetails(await getClientParcelsDetails(clientId));
```

Comment:
If `displayClientsParcels` is false, should this be called?

Reply:
This is false in ParcelForm.tsx. 

I think in that case `clientParcelsDetails` should be left as null.